src/components/refactor/dialogs/two-windings-transformer-creation/tap-changer-pane/tap-changer-pane-taps.js

```diff
@@ -0,0 +1,368 @@
+                disabled: disabled,
+            }}
+        />
+    );
+
+    const createRuleButton = (
+        <Tooltip
+            title={intl.formatMessage({
+                id: createRuleMessageId,
+            })}
+            placement="left"
+        >
+            <span>
+                <IconButton
+                    onClick={() => setOpenCreateRuleDialog(true)}
+                    disabled={disabled || tapFields.length === 0}
+                    sx={{
+                        position: 'absolute',
+                        right: 0,
+                        top: 0,
+                        padding: 1.5,
+                    }}
+                >
+                    <AddchartIcon />
+                </IconButton>
+            </span>
+        </Tooltip>
+    );
+
+    const completedColumnsDefinition = columnsDefinition.map((column, index) =>
+        index === columnsDefinition.length - 1
+            ? { ...column, extra: createRuleButton }
+            : column
+    );
+
+    return (
+        <Grid item container spacing={1}>
+            <Grid item container spacing={2}>
+                <Grid item xs={4}>
+                    {lowTapPositionField}
+                </Grid>
+                <Grid item xs={4}>
+                    {highTapPositionField}
+                </Grid>
+                <Grid item xs={4}>
+                    {tapPositionField}
+                </Grid>
+            </Grid>
+            <DndTable
+                arrayFormName={`${tapChanger}.${STEPS}`}
+                useFieldArrayOutput={useFieldArrayOutput}
+                disabled={disabled}
+                columnsDefinition={completedColumnsDefinition}
+                handleAddButton={handleAddRowsButton}
+                handleDeleteButton={deleteSelectedRows}
+                handleUploadButton={handleImportTapRuleButton}
+                uploadButtonMessageId={importRuleMessageId}
+            />
+            <CreateRuleDialog
+                ruleType={ruleType}
+                openCreateRuleDialog={openCreateRuleDialog}
+                setOpenCreateRuleDialog={setOpenCreateRuleDialog}
+                handleCreateTapRule={handleCreateTapRule}
+                allowNegativeValues={false}
+            />
+            <ImportRuleDialog
+                ruleType={ruleType}
+                openImportRuleDialog={openImportRuleDialog}
+                setOpenImportRuleDialog={setOpenImportRuleDialog}
+                csvColumns={csvColumns}
+                handleImportTapRule={handleImportTapRule}
+            />
+            <AddRowsDialog
+                open={openAddRowsDialog}
+                handleAddRowsButton={addNewRows}
+                onClose={handleCloseAddRowsDialog}
+            />
+        </Grid>
+    );
+};
+
+TapChangerPaneTaps.prototype = {
+    tapChanger: PropTypes.string.isRequired,
+    ruleType: PropTypes.string.isRequired,
+    createTapRuleColumn: PropTypes.string.isRequired,
+    disabled: PropTypes.bool,
```

Comment:
is disabled prop required ?

Reply:
For it's not required because the component can works without it.